[WebTransport] Support latest HTTP datagram

yutakahirano opened the pull request:
Until the protocol is formally implemented by aioquic, we need to talk to
clients talking with the protocol. Let's add some ad-hoc support.

yutakahirano committed:
[WIP][WebTransport] Support the latest HTTP Datagram draft

yutakahirano committed:
fix

yutakahirano committed:
fix

yutakahirano commented:
@nidhijaju @vasilvv can you take a look?

vasilvv commented on tools/webtransport/h3/webtransport_h3_server.py:
```diff
@@ -72,6 +72,38 @@ def encode(self) -> bytes:
         return buffer.data
 
 
+class MyH3Connection(H3Connection):
```
We should probably have a more descriptive name for this, unless the "My" prefix is a convention.

yutakahirano replied:
Done.

vasilvv reviewed (approved):
Overall LGTM. Could you add a check that makes sure we don't receive capsules when DRAFT04 is not supported?

nidhijaju reviewed (approved):
I wasn't able to test that this works in my local environment as I still only see the "Opening handshake failed."/unresponsive page error when running the datagrams WPTs, but logic-wise looks okay to me.

yutakahirano committed:
fix

yutakahirano commented:
> Could you add a check that makes sure we don't receive capsules when DRAFT04 is not supported?

I'll add the check to #30982.

yutakahirano committed:
fix

yutakahirano committed:
fix

yutakahirano committed:
fix

yutakahirano commented:
>> Could you add a check that makes sure we don't receive capsules when DRAFT04 is not supported?
>
> I'll add the check to #30982.

I changed my mind, this is now addressed.

yutakahirano committed:
fix

yutakahirano committed:
fix

yutakahirano committed:
fix

yutakahirano commented:
Hi @web-platform-tests/wpt-core-team, I'd like to ask for your help. This change is blocked on some flaky tests (see [this](https://github.com/web-platform-tests/wpt/pull/31049/checks?check_run_id=3768302024) for example), but that's weird. We haven't run WebTransport server on CI bots so the WPT server is running without WebTransport over HTTP/3 support, and in that experiment I'm expect the tests (e.g.,. `/webtransport/datagrams.https.any.serviceworker.html`) fail consistently, which matches with my local results. 

Thanks!

yutakahirano committed:
fix

yutakahirano committed:
fix

yutakahirano committed:
fix

yutakahirano committed:
fix

yutakahirano commented:
``

yutakahirano committed:
add metadata

yutakahirano committed:
fix

yutakahirano committed:
fix

yutakahirano commented:
Disabling (some) webtransport/ tests with .ini files.
